idb/postgres/postgres.go

```diff
@@ -790,25 +790,41 @@ func (db *IndexerDb) yieldTxnsThreadSimple(ctx context.Context, rows pgx.Rows, r
 		var round uint64
 		var asset uint64
 		var intra int
-		var txnbytes []byte
-		var roottxnbytes []byte
+		var txn []byte
+		var roottxn []byte
 		var extraJSON []byte
 		var roundtime time.Time
-		err := rows.Scan(&round, &intra, &txnbytes, &roottxnbytes, &extraJSON, &asset, &roundtime)
+		err := rows.Scan(&round, &intra, &txn, &roottxn, &extraJSON, &asset, &roundtime)
 		var row idb.TxnRow
 		if err != nil {
 			row.Error = err
 		} else {
 			row.Round = round
 			row.Intra = intra
-			row.TxnBytes = txnbytes
-			row.RootTxnBytes = roottxnbytes
+			if roottxn != nil {
+				// Inner transaction.
+				row.RootTxn = new(transactions.SignedTxnWithAD)
+				*row.RootTxn, err = encoding.DecodeSignedTxnWithAD(roottxn)
+				if err != nil {
+					err = fmt.Errorf("error decoding roottxn, err: %w", err)
+					row.Error = err
+				}
+			} else {
```

Comment:
nit: this would make it easier to support returning inner transactions in the future.
```suggestion
			}
			if {
```